components/builder-depot/src/handlers/integrations.rs

```diff
@@ -74,6 +71,32 @@ pub fn validate_params(
     Ok(res)
 }
 
+pub fn fetch_origin_integrations(req: &mut Request) -> IronResult<Response> {
+    let params = match validate_params(req, &["origin"]) {
+        Ok(p) => p,
+        Err(st) => return Ok(Response::with(st)),
+    };
+    let mut request = OriginIntegrationRequest::new();
+    request.set_origin(params["origin"].clone());
+    match route_message::<OriginIntegrationRequest, OriginIntegrationResponse>(req, &request) {
+        Ok(oir) => {
+            println!("{:?}", &oir.get_integrations());
+            let integrations_response: HashMap<String, Vec<String>> = oir.get_integrations()
+                .iter()
+                .fold(HashMap::new(), |mut acc, ref i| {
+                    acc.entry(i.get_integration().to_owned())
+                        .or_insert(Vec::new())
+                        .push(i.get_name().to_owned());
+                    acc
+                });
```

Comment:
This is a terrific little snippet of code.  The `.entry().or_insert().push()` trick is one I just recently learned for populating a hash where the values are Vec's.

Reply:
🎉